solr/core/src/java/org/apache/solr/jersey/InjectionFactories.java

```diff
@@ -104,6 +106,24 @@ public IndexSchema provide() {
     public void dispose(IndexSchema instance) {}
   }
 
+  public static class ReuseFromContextSolrParamsFactory implements Factory<SolrParams> {
```

Comment:
[+1] Technically, v2 code that needs the SolrParams could inject the SolrQueryRequest and call `getParams`.  So this injector isn't strictly required.

But I'm a big fan of only injecting what you actually need - it makes the code easier to understand, and really helps simplify unit tests.  So 👍